parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -1000,6 +1042,21 @@ public ReadBuilder withNameMapping(NameMapping newNameMapping) {
       return this;
     }
 
+    public ReadBuilder withKeyMetadata(ByteBuffer encryptionKeyMetadata) {
```

Comment:
This is a ByteBuffer but the other was a class. That doesn't seem quite right.

Reply:
Per the discussion above, we'll have only one type of key metadata. So replacing ByteBuffer with that class should be fine.